piptools/exceptions.py

```diff
@@ -13,6 +13,10 @@ def __str__(self):
             'Could not find a version that matches {}'.format(self.ireq),
             'Tried: {}'.format(', '.join(str(version) for version in sorted_versions) or '(no version found at all)')
         ]
+        if sorted_versions:
+            lines.append('There are incompatible versions in the resolved dependencies.')
+        else:
+            lines.append('Was the PyPi repository reachable?')
```

Comment:
We know which pypi repository was tried, I would add it here. If it is an internal server, this will help diagnose errors vs public pypi. Small nit:

`Is the $(pypi) repo reachable?`

Reply:
There we go